### PR TITLE
Added utility classes 

### PR DESCRIPTION
**Pull request by a-panchenko**
Added utility classes which are responsible for converting com.google.cloud.datastore.Entity to objects of the specified type and vice verse.

### CODE REVIEW

File: src/main/scala/io/applicative/datastore/util/reflection/Field.scala [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package io.applicative.datastore.util.reflection
+
+case class Field[T >: Any](name: String, value: T)
```
Comment (danosipov) [inline src/main/scala/io/applicative/datastore/util/reflection/Field.scala:3]: Is `[T >: Any]` just the same as `[T]`?

File: src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  private[datastore] def extractRuntimeClass[E: TypeTag](): RuntimeClass = {
+    val mirror = runtimeMirror(getClass.getClassLoader)
+    mirror.runtimeClass(typeOf[E].typeSymbol.asClass)
+  }
+
+  private[datastore] def instanceToDatastoreEntity[E](key: Key[E], classInstance: E, clazz: Class[_]): Entity = {
+    var builder = Entity.newBuilder(key.key)
+    clazz.getDeclaredFields
+      .filterNot(_.isSynthetic)
+      .tail
```
Comment (danosipov) [inline src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala]: What's the reason `head` is not processed? Can you add a comment?

Comment (a-panchenko) [reply src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala]: I assume that the first field is ID, which is already encapsulated in Key, so there is no need to add id to the builder.

File: src/main/scala/io/applicative/datastore/util/reflection/Field.scala (hunk 1 above)
Comment (a-panchenko) [reply src/main/scala/io/applicative/datastore/util/reflection/Field.scala:3]: [T] is treated as [T >: AnyRef] while Long, Byte, Int and other wrappers of primitive types are successors of AnyVal.

Comment (danosipov) [reply src/main/scala/io/applicative/datastore/util/reflection/Field.scala:3]: Got it - thanks!

File: src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala (hunk 2 above)
Comment (danosipov) [reply src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala]: Cool - can you add a comment that explains this?

Comment (a-panchenko) [reply src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala]: @danosipov Done.

**Comment by danosipov** [comment]
👍

**Pull request merged**